### PR TITLE
Support negative observations

### PR DESCRIPTION
**Pull request by connorwstein**

### CODE REVIEW

File: pkg/terra/marshal_signed_int.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+		val.Sub(maxUint, val)
+		val.Neg(val)
+	}
+	return val, nil
+}
+
+// ToBytes converts *big.Int o into bytes as a big-endian signed
+// integer of size numBytes
+func ToBytes(o *big.Int, numBytes uint) ([]byte, error) {
+	negative := o.Sign() < 0
```
Comment (aalu1418) [inline pkg/terra/marshal_signed_int.go:33]: should this have validation that the big.Int fits in the expected number of bytes?

like in libocr: https://github.com/smartcontractkit/libocr/blob/449836e6cfec324fbd6781e9a4d4b953840cdf0b/offchainreporting2/reportingplugin/median/observation.go#L21

File: pkg/terra/marshal_signed_int.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+// ToInt interprets bytes s as a big-endian signed integer
+// of size numBytes.
+func ToInt(s []byte, numBytes uint) (*big.Int, error) {
+	if uint(len(s)) != numBytes {
+		return nil, fmt.Errorf("invalid int length: expected %d got %d", numBytes, len(s))
+	}
+	val := (&big.Int{}).SetBytes(s)
+	numBits := numBytes * 8
+	// 2**(numBits-1) - 1
+	maxPositive := big.NewInt(0).Sub(big.NewInt(0).Lsh(big.NewInt(1), numBits-1), big.NewInt(1))
```
Comment (aalu1418) [inline pkg/terra/marshal_signed_int.go:18]: could separate this into another helper func so we can use it in both functions

**Review by aalu1418 — CHANGES_REQUESTED** [review]
Also just a general question: where should we store this helper?

if we wanted to use this in chainlink-solana...would the imports get weird?

**Review by jmank88 — DISMISSED** [review]

**Comment by connorwstein** [comment]
> Also just a general question: where should we store this helper?
> 
> if we wanted to use this in chainlink-solana...would the imports get weird?

I'll try to just PR this to OCR

**Review by aalu1418 — APPROVED** [review]
🎉

**Pull request merged**